packages/builder/src/components/automation/SetupPanel/AutomationBlockSetup.svelte

```diff
@@ -462,27 +462,75 @@
               <div class="label-wrapper">
                 <Label>{label}</Label>
               </div>
-              <div class="attachment-field-width">
-                <KeyValueBuilder
-                  on:change={e =>
+              <div class="toggle-container">
+                <Toggle
+                  value={inputData?.meta?.useAttachmentBinding}
+                  text={"Use bindings"}
+                  size={"XS"}
+                  on:change={e => {
                     onChange(
                       {
-                        detail: e.detail.map(({ name, value }) => ({
-                          url: name,
-                          filename: value,
-                        })),
+                        detail: "",
                       },
                       key
-                    )}
-                  object={handleAttachmentParams(inputData[key])}
-                  allowJS
-                  {bindings}
-                  keyBindings
-                  customButtonText={"Add attachment"}
-                  keyPlaceholder={"URL"}
-                  valuePlaceholder={"Filename"}
+                    )
+                    onChange(
+                      { detail: { useAttachmentBinding: e.detail } },
+                      "meta"
+                    )
+                  }}
                 />
               </div>
+
+              <div class="attachment-field-width">
+                {#if !inputData?.meta?.useAttachmentBinding}
+                  <KeyValueBuilder
+                    on:change={e =>
+                      onChange(
+                        {
+                          detail: e.detail.map(({ name, value }) => ({
+                            url: name,
+                            filename: value,
+                          })),
+                        },
+                        key
+                      )}
+                    object={handleAttachmentParams(inputData[key])}
+                    allowJS
+                    {bindings}
+                    keyBindings
+                    customButtonText={"Add attachment"}
+                    keyPlaceholder={"URL"}
+                    valuePlaceholder={"Filename"}
+                  />
+                {:else if isTestModal}
+                  <ModalBindableInput
+                    title={value.title || label}
+                    value={inputData[key]}
+                    panel={AutomationBindingPanel}
+                    type={value.customType}
+                    on:change={e => onChange(e, key)}
+                    {bindings}
+                    updateOnChange={false}
+                  />
+                {:else}
+                  <div class="test">
```

Comment:
Is this "test" class expected?

Reply:
Yep in this case that's the class for the "test" drawer. It's also used elsewhere in that file